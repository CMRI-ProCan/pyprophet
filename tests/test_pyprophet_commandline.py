import os
import subprocess
import shutil
import sys

import pandas as pd

import pytest


d = pd.options.display
d.width = 220
d.precision = 6

DATA_FOLDER = os.path.join(os.path.dirname(os.path.abspath(__file__)), "data")


def _record(stdout, regtest):
    for line in stdout.split("\n"):
        if line.startswith("INFO"):
            # remove data which is not constant over runs as time or process id:
            line = line.split(":", 4)[-1]
        if "TIME" in line:
            continue
        print >> regtest, line


def _expected_output_files():
    return ["test_data_ms2_scored.tsv",
            "test_data_ms2_report.pdf",
            "test_data_ms2_weights.csv",
            "test_data_ms2_summary_stat.csv",
            "test_data_ms2_full_stat.csv",
            ]


def _remove_output_files(tmpdir, names=None):
    if names is None:
        names = _expected_output_files()
    for name in names:
        full_path = os.path.join(tmpdir.strpath, name)
        os.remove(full_path)


def _dump_output_files(tmpdir, regtest, names=None):
    try:
        tmpdir = tmpdir.strpath
    except AttributeError:
        pass
    if names is None:
        names = _expected_output_files()
    for name in names:
        full_path = os.path.join(tmpdir, name)
        if not name.endswith(".pdf"):
            _dump(full_path, regtest)


def _dump(full_path, regtest):
    head = open(full_path).readline()
    if "\t" in head:
        df = pd.read_csv(full_path, sep="\t", header=None)
    else:
        df = pd.read_csv(full_path, sep=" ", header=None)

    # lines = open(full_path, "r").readlines()
    # f = os.path.basename(full_path)
    if len(df) > 20:
        print >> regtest, df[:10]
        print >> regtest, "..."
        print >> regtest, df[-10:]
    else:
        print >> regtest, df


def _run_cmdline(cmdline):
    stdout = cmdline + "\n"
    try:
        stdout += subprocess.check_output(cmdline, shell=True,
                                          stderr=subprocess.STDOUT)
    except subprocess.CalledProcessError, e:
        print >> sys.stderr, e.output
        raise
    return stdout


def _run_pyprophet_to_learn_model(regtest, temp_folder, dump_result_files=False, parametric=False, pfdr=False, pi0_lambda=False):
    os.chdir(temp_folder)
    data_path = os.path.join(DATA_FOLDER, "test_data.txt")
    shutil.copy(data_path, temp_folder)
    cmdline = "pyprophet score --in=test_data.txt --test"
    if parametric:
        cmdline += " --parametric"
    if pfdr:
        cmdline += " --pfdr"
    if pi0_lambda is not False:
        cmdline += " --pi0_lambda=" + pi0_lambda
    stdout = _run_cmdline(cmdline)

    for f in _expected_output_files():
        full_path = os.path.join(temp_folder, f)
        assert os.path.exists(full_path)

    if not dump_result_files:
        return stdout

    _dump_output_files(temp_folder, regtest)

    return stdout


def test_tsv_0(tmpdir, regtest):
    stdout = _run_pyprophet_to_learn_model(regtest, tmpdir.strpath, True)
    _record(stdout, regtest)

def test_tsv_1(tmpdir, regtest):
    stdout = _run_pyprophet_to_learn_model(regtest, tmpdir.strpath, True, True)
    _record(stdout, regtest)

def test_tsv_2(tmpdir, regtest):
    stdout = _run_pyprophet_to_learn_model(regtest, tmpdir.strpath, True, False, True)
    _record(stdout, regtest)

def test_tsv_3(tmpdir, regtest):
    stdout = _run_pyprophet_to_learn_model(regtest, tmpdir.strpath, True, False, False, "0.3 0.55 0.05")
    _record(stdout, regtest)

def test_tsv_apply_weights(tmpdir, regtest):

    stdout = _run_pyprophet_to_learn_model(regtest, tmpdir.strpath, True)

    _record(stdout, regtest)

    full_path = os.path.join(tmpdir.strpath, "test_data_ms2_scored.tsv")
    _dump(full_path, regtest)

    # collect q-value stats
    m_score_stat, = [l for l in stdout.split("\n") if "mean q_value" in l]

    # split away log time etc:
    __, __, interesting_m_score_output = m_score_stat.partition("mean q_value")

    # collect s value stats
    s_value_stat, = [l for l in stdout.split("\n") if "mean s_value" in l]

    # split away log time etc:
    __, __, interesting_s_value_output = s_value_stat.partition("mean s_value")

    stdout = _run_cmdline("pyprophet score --in=test_data.txt --apply_weights=test_data_ms2_weights.csv "
                          "--test")

    _record(stdout, regtest)

    # collect m score stats
    m_score_stat, = [l for l in stdout.split("\n") if "mean q_value" in l]

    # split away log time etc:
    __, __, interesting_m_score_output2 = m_score_stat.partition("mean q_value")

    # collect s value stats
    s_value_stat, = [l for l in stdout.split("\n") if "mean s_value" in l]

    # split away log time etc:
    __, __, interesting_s_value_output2 = s_value_stat.partition("mean s_value")

    # check if output is consistent over runs:
    assert interesting_m_score_output == interesting_m_score_output2
    assert interesting_s_value_output == interesting_s_value_output2


def test_not_unique_tg_id_blocks(tmpdir):

    os.chdir(tmpdir.strpath)
    data_path = os.path.join(DATA_FOLDER, "test_invalid_data.txt")
    shutil.copy(data_path, tmpdir.strpath)
    cmdline = "pyprophet score --in=test_invalid_data.txt --test"

    with pytest.raises(subprocess.CalledProcessError) as exc_info:
        subprocess.check_output(cmdline, shell=True, stderr=subprocess.STDOUT)

    e = exc_info.value
    assert "Error: group_id values do not form unique blocks in input file(s)." in e.output
