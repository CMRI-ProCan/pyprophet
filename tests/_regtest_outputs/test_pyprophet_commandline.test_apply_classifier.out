         0       1      2                               3                  4                            5                    6                              7                          8                 9   \
0  group_id  run_id  decoy  main_var_xx_swath_prelim_score  var_bseries_score  var_elution_model_fit_score  var_intensity_score  var_isotope_correlation_score  var_isotope_overlap_score  var_library_corr
1  459_run0       0      0                   5.27893963303                  5               0.993098149697       0.930480707464                 0.998724247981                        0.0     0.91240602174
2  459_run0       0      0                  -0.13974375122                  0               0.953924566507      0.0259317385401                 0.632511052964                        0.0   -0.285526965432
3  459_run0       0      0                  -0.77504285962                  0               0.666666249434     0.00044235703213                 0.657922811238                        0.0    0.376474234306
4  459_run0       0      0                  -1.05454511848                  0               0.847340305646      0.0009039469787                  0.79843565088             0.255319140851     0.51872168654
5  459_run0       0      0                    -1.246166932                  0                         -1.0     0.00042312411769                 0.563846670955            0.0454545468092    0.445906014642
6  459_run0       0      0                  -1.27951627249                  1               0.931647390127     0.00561216443373               -0.0419896690886            0.0997258424759  -0.0976738454584
7  459_run0       0      0                  -1.81518747764                  0               0.933585554361     0.00048082286101                 0.104774965663             0.159999996424   -0.630422997778
8  460_run0       0      0                   4.90480697454                  0               0.991616976261       0.929647682861                 0.998396899384                        0.0    0.757169232785
9  460_run0       0      0                   1.00012878546                  0               0.836714565754     0.00222462693417                 0.762015540501             0.380021706223    0.956728181447

        ...                          15                            16               17                        18                 19               20                 21                 22                 23  \
0       ...         var_xcorr_coelution  var_xcorr_coelution_weighted  var_xcorr_shape  var_xcorr_shape_weighted  var_yseries_score          d_score            p_value            q_value                pep
1       ...                         0.0                           0.0   0.960350100659            0.970979341002                  6    5.50973510742  1.79687089652e-08  2.35192525723e-09  1.08601512169e-08
2       ...               1.39668108475                0.859180469403   0.789175310515            0.716396888252                  0    -1.0907715559     0.860200595908    0.0560026429628                1.0
3       ...               2.38678570295                 1.97001683336   0.119047619048           0.0149915833201                  0    -4.5365819931     0.979526073734    0.0632768781482                1.0
4       ...               4.39320066984                 2.78930134498   0.329194873727            0.330811834697                  0   -3.23349261284     0.979526073734    0.0632768781482                1.0
5       ...                         1.0                           1.0              0.0                       0.0                  0   -1.98865056038     0.979526073734    0.0632768781482                1.0
6       ...               2.20848900055                 1.49086544255   0.711876219914            0.685971015368                  1   -3.35698604584     0.979526073734    0.0632768781482                1.0
7       ...               1.39696341001                0.799699939913   0.761807409832            0.743328859302                  0   -4.94960355759     0.979526073734    0.0632768781482                1.0
8       ...               2.17321312465                 1.00181054984   0.895313998806            0.867887743561                  5    4.87290096283  5.49856905785e-07  5.28708563255e-08  6.51766733095e-07
9       ...               3.54384793756                 1.31714642152   0.666435350514            0.636692707326                  0  -0.428127616644      0.67651554102    0.0446250356873                1.0

                24
0  peak_group_rank
1                1
2                2
3                6
4                4
5                3
6                5
7                7
8                1
9                2

[10 rows x 25 columns]
...
                  0  1  2                3  4               5                 6                7                8                 9  ...             15              16              17              18 19  \
9156  DECOY_592_run0  0  1  -0.909754229584  0  0.823102881511   0.0126013214359  -0.147306602511   0.229729726911    0.778375379467 ...  3.94538540661   2.00615080523  0.604014629053  0.541176149999  2
9157  DECOY_592_run0  0  1  -0.956865429723  0            -1.0  0.00238403378516   0.266941597725              0.0  -0.0969535542793 ...            1.0             1.0             0.0             0.0  0
9158  DECOY_592_run0  0  1   -1.52418394133  0            -1.0  0.00017028812751  -0.583329727443              0.0  -0.0585549720413 ...            1.0             1.0             0.0             0.0  0
9159  DECOY_592_run0  0  1   -1.68998208875  0  0.895950257778  0.00854846400109   0.546651481853   0.601593613625   -0.189017878466 ...  2.06934963933   1.14626069677  0.785186892817  0.764795972809  0
9160  DECOY_592_run0  0  1     -1.692134624  0  0.958494941394  0.00630066071793  -0.328751834466   0.189189195633   -0.482215863506 ...  1.28809100151  0.791833291782  0.803144337956  0.768010655265  0
9161  DECOY_592_run0  0  1   -1.84945466301  0  0.633835911751  0.00561950820789   0.371593955459              0.0   -0.371771330281 ...  2.61135263418   2.02039942856   0.55597268955  0.330880632506  0
9162  DECOY_592_run0  0  1   -2.10957129817  0  0.906453510125  0.00527893195286    0.22376248428  0.0645161271095   -0.255514614107 ...  1.07777568371  0.598394985285  0.893420898864  0.890335674837  0
9163  DECOY_592_run0  0  1   -2.31560973872  0  0.905198852221   0.0028948981677   0.137664821691              0.0    0.876978922737 ...  3.43237660856   2.17985387492  0.539929223558   0.44766613662  0
9164  DECOY_592_run0  0  1   -2.53264045217  0  0.980351785819  0.00664123697296  0.0745622896143   0.205128207803    0.216855907728 ...  1.87132796282  0.908604380766  0.736385791197  0.708796452491  0
9165  DECOY_592_run0  0  1   -3.44062881815  0            -1.0  0.00102172876507  0.0339463918184   0.333333343267   0.0837371951856 ...            1.0             1.0             0.0             0.0  0

                  20              21               22   23  24
9156  -3.49780750275  0.979526073734  0.0632768781482  1.0  11
9157  -1.00938773155  0.860200595908  0.0560026429628  1.0   2
9158  -4.97128486633  0.979526073734  0.0632768781482  1.0  14
9159  -4.11790084839  0.979526073734  0.0632768781482  1.0  12
9160  -2.77628207207  0.979526073734  0.0632768781482  1.0   9
9161  -1.98261642456  0.979526073734  0.0632768781482  1.0   5
9162  -3.34569406509  0.979526073734  0.0632768781482  1.0  10
9163  -5.20470523834  0.979526073734  0.0632768781482  1.0  16
9164  -5.07580375671  0.979526073734  0.0632768781482  1.0  15
9165  -4.78132820129  0.979526073734  0.0632768781482  1.0  13

[10 rows x 25 columns]
                                                    0
0                                  score,weight,level
1   main_var_xx_swath_prelim_score,0.117474541815,ms2
2               var_bseries_score,-0.111879898204,ms2
3      var_elution_model_fit_score,-2.12447173592,ms2
4              var_intensity_score,0.887394406749,ms2
5      var_isotope_correlation_score,1.9326851711,ms2
6        var_isotope_overlap_score,-2.00679092253,ms2
7                var_library_corr,-0.519988334781,ms2
8                 var_library_rmsd,-10.9132137214,ms2
9               var_log_sn_score,-0.0494314992607,ms2
10               var_massdev_score,0.032629600868,ms2
11    var_massdev_score_weighted,-0.0690910447715,ms2
12               var_norm_rt_score,-19.8656343168,ms2
13             var_xcorr_coelution,0.228202040922,ms2
14   var_xcorr_coelution_weighted,-0.411326493739,ms2
15                  var_xcorr_shape,3.99892628169,ms2
16         var_xcorr_shape_weighted,0.62380530872,ms2
17               var_yseries_score,0.214296627148,ms2
                                                   0
0  qvalue,pvalue,svalue,pep,fdr,fnr,fpr,tp,tn,fp,...
1  0.0,2.0872192863e-13,0.0165745856353,6.2484568...
2  0.00999999977648,0.13655127418,1.0,0.627174477...
3  0.019999999553,0.301996778987,1.0,1.0,0.020350...
4  0.0500000007451,0.763397639012,1.0,1.0,0.04983...
5  0.10000000149,0.979526073734,1.0,1.0,0.0632768...
6                           0.20000000298,,,,,,,,,,,
7                          0.300000011921,,,,,,,,,,,
8                           0.40000000596,,,,,,,,,,,
9                                     0.5,,,,,,,,,,,
                                                   0
0  cutoff,fdr,fn,fnr,fp,fpr,pvalue,qvalue,svalue,...
1  -2.51781249046,0.0632768781482,-0.511848156649...
2  -2.30936026573,0.0632768781482,-0.511848156649...
3  -2.10090827942,0.0632768781482,-0.511848156649...
4  -1.89245605469,0.0632768781482,-0.511848156649...
5  -1.68400382996,0.0602653598684,-0.73757109081,...
6  -1.47555172443,0.0602653598684,-0.73757109081,...
7  -1.2670994997,0.0594883540255,-0.0969837001802...
8  -1.05864727497,0.0560026429628,-0.494985102301...
9  -0.850195169449,0.0498301330948,-1.9150590247,...
...
                                                    0
42  6.02872657776,1.81822701645e-10,253.00000002,0...
43  6.23717880249,6.89725360159e-11,282.000000006,...
44  6.44563102722,3.05631677374e-11,314.000000001,...
45  6.65408325195,1.14508760896e-11,331.0,0.929775...
46  6.86253547668,4.08839628818e-12,344.0,0.932249...
47  7.07098770142,1.72589215646e-12,351.0,0.933510...
48  7.27943944931,1.1407541578e-12,358.0,0.9347258...
49  7.48789167404,1.34614541736e-12,361.0,0.935233...
50  7.69634389877,1.34614541736e-12,361.0,0.935233...
51  7.9047961235,1.34614541736e-12,361.0,0.9352331...
pyprophet score --in=test_data.txt --random_seed=42
 config settings:
     final_statistics.emp_p: False
     final_statistics.lambda: 0.4
     final_statistics.lfdr_adj: 1.5
     final_statistics.lfdr_eps: 1e-08
     final_statistics.lfdr_monotone: True
     final_statistics.lfdr_transf: probit
     final_statistics.lfdr_trunc: True
     final_statistics.pfdr: False
     final_statistics.pi0_method: smoother
     final_statistics.pi0_smooth_df: 3
     final_statistics.pi0_smooth_log_pi0: False
     group_id: group_id
     is_test: False
     num_processes: 1
     random_seed: 42
     semi_supervised_learner.initial_fdr: 0.15
     semi_supervised_learner.initial_lambda: 0.4
     semi_supervised_learner.iteration_fdr: 0.02
     semi_supervised_learner.iteration_lambda: 0.4
     semi_supervised_learner.num_iter: 10
     tric_chromprob: False
     xeval.fraction: 0.5
     xeval.num_iter: 10
 learn and apply classifier from input data
 Data set contains 387 decoy and 387 target groups.
 summary input file:
    9165 lines
    774 transition groups
    17 scores including main score
 learn and apply scorer
 start 10 cross evals using 1 processes
 start learn_randomized
 end learn_randomized
 start learn_randomized
 end learn_randomized
 start learn_randomized
 end learn_randomized
 start learn_randomized
 end learn_randomized
 start learn_randomized
 end learn_randomized
 start learn_randomized
 end learn_randomized
 start learn_randomized
 end learn_randomized
 start learn_randomized
 end learn_randomized
 start learn_randomized
 end learn_randomized
 start learn_randomized
 end learn_randomized
 finished cross evals

 Data set contains 387 decoy and 387 target groups.
 mean q_value = 5.515478e-02, std_dev q_value = 1.675134e-02
 mean s_value = 9.795502e-01, std_dev s_value = 1.132123e-01
 calculated scoring and statistics
 processing input data finished
==================================================================================================
   qvalue        pvalue    svalue           pep           fdr       fnr           fpr          tp  \
0    0.00  2.087219e-13  0.016575  6.248457e-09  8.696747e-13  0.934383  2.087219e-13    6.000000
1    0.01  1.365513e-01  1.000000  6.271745e-01  9.301858e-03  0.000000  1.365513e-01  363.586218
2    0.02  3.019968e-01  1.000000  1.000000e+00  2.035019e-02  0.000000  3.019968e-01  363.450081
3    0.05  7.633976e-01  1.000000  1.000000e+00  4.983013e-02  0.000000  7.633976e-01  363.915059
4    0.10  9.795261e-01  1.000000  1.000000e+00  6.327688e-02  0.000000  9.795261e-01  362.511848
5    0.20           NaN       NaN           NaN           NaN       NaN           NaN         NaN
6    0.30           NaN       NaN           NaN           NaN       NaN           NaN         NaN
7    0.40           NaN       NaN           NaN           NaN       NaN           NaN         NaN
8    0.50           NaN       NaN           NaN           NaN       NaN           NaN         NaN

          tn            fp          fn    cutoff
0  25.000000  5.218048e-12  356.000000  7.249766
1  21.586218  3.413782e+00   -1.586218  1.095946
2  17.450081  7.549919e+00   -1.450081  0.518666
3   5.915059  1.908494e+01   -1.915059 -0.717274
4   0.511848  2.448815e+01   -0.511848 -2.044058
5        NaN           NaN         NaN       NaN
6        NaN           NaN         NaN       NaN
7        NaN           NaN         NaN       NaN
8        NaN           NaN         NaN       NaN
==================================================================================================
WRITTEN: test_data_ms2_summary_stat.csv
WRITTEN: test_data_ms2_full_stat.csv
WRITTEN: test_data_ms2_scored.tsv
WRITTEN: test_data_ms2_report.pdf
WRITTEN: test_data_ms2_weights.csv
TIME: 0 seconds and 838 msecs wall time

         0       1      2                               3                  4                            5                    6                              7                          8                 9   \
0  group_id  run_id  decoy  main_var_xx_swath_prelim_score  var_bseries_score  var_elution_model_fit_score  var_intensity_score  var_isotope_correlation_score  var_isotope_overlap_score  var_library_corr
1  459_run0       0      0                   5.27893963303                  5               0.993098149697       0.930480707464                 0.998724247981                        0.0     0.91240602174
2  459_run0       0      0                  -0.13974375122                  0               0.953924566507      0.0259317385401                 0.632511052964                        0.0   -0.285526965432
3  459_run0       0      0                  -0.77504285962                  0               0.666666249434     0.00044235703213                 0.657922811238                        0.0    0.376474234306
4  459_run0       0      0                  -1.05454511848                  0               0.847340305646      0.0009039469787                  0.79843565088             0.255319140851     0.51872168654
5  459_run0       0      0                    -1.246166932                  0                         -1.0     0.00042312411769                 0.563846670955            0.0454545468092    0.445906014642
6  459_run0       0      0                  -1.27951627249                  1               0.931647390127     0.00561216443373               -0.0419896690886            0.0997258424759  -0.0976738454584
7  459_run0       0      0                  -1.81518747764                  0               0.933585554361     0.00048082286101                 0.104774965663             0.159999996424   -0.630422997778
8  460_run0       0      0                   4.90480697454                  0               0.991616976261       0.929647682861                 0.998396899384                        0.0    0.757169232785
9  460_run0       0      0                   1.00012878546                  0               0.836714565754     0.00222462693417                 0.762015540501             0.380021706223    0.956728181447

        ...                          15                            16               17                        18                 19               20                 21                 22                 23  \
0       ...         var_xcorr_coelution  var_xcorr_coelution_weighted  var_xcorr_shape  var_xcorr_shape_weighted  var_yseries_score          d_score            p_value            q_value                pep
1       ...                         0.0                           0.0   0.960350100659            0.970979341002                  6    5.50973510742  1.79687089652e-08  2.35192525723e-09  1.08601512169e-08
2       ...               1.39668108475                0.859180469403   0.789175310515            0.716396888252                  0    -1.0907715559     0.860200595908    0.0560026429628                1.0
3       ...               2.38678570295                 1.97001683336   0.119047619048           0.0149915833201                  0    -4.5365819931     0.979526073734    0.0632768781482                1.0
4       ...               4.39320066984                 2.78930134498   0.329194873727            0.330811834697                  0   -3.23349261284     0.979526073734    0.0632768781482                1.0
5       ...                         1.0                           1.0              0.0                       0.0                  0   -1.98865056038     0.979526073734    0.0632768781482                1.0
6       ...               2.20848900055                 1.49086544255   0.711876219914            0.685971015368                  1   -3.35698604584     0.979526073734    0.0632768781482                1.0
7       ...               1.39696341001                0.799699939913   0.761807409832            0.743328859302                  0   -4.94960355759     0.979526073734    0.0632768781482                1.0
8       ...               2.17321312465                 1.00181054984   0.895313998806            0.867887743561                  5    4.87290096283  5.49856905785e-07  5.28708563255e-08  6.51766733095e-07
9       ...               3.54384793756                 1.31714642152   0.666435350514            0.636692707326                  0  -0.428127616644      0.67651554102    0.0446250356873                1.0

                24
0  peak_group_rank
1                1
2                2
3                6
4                4
5                3
6                5
7                7
8                1
9                2

[10 rows x 25 columns]
...
                  0  1  2                3  4               5                 6                7                8                 9  ...             15              16              17              18 19  \
9156  DECOY_592_run0  0  1  -0.909754229584  0  0.823102881511   0.0126013214359  -0.147306602511   0.229729726911    0.778375379467 ...  3.94538540661   2.00615080523  0.604014629053  0.541176149999  2
9157  DECOY_592_run0  0  1  -0.956865429723  0            -1.0  0.00238403378516   0.266941597725              0.0  -0.0969535542793 ...            1.0             1.0             0.0             0.0  0
9158  DECOY_592_run0  0  1   -1.52418394133  0            -1.0  0.00017028812751  -0.583329727443              0.0  -0.0585549720413 ...            1.0             1.0             0.0             0.0  0
9159  DECOY_592_run0  0  1   -1.68998208875  0  0.895950257778  0.00854846400109   0.546651481853   0.601593613625   -0.189017878466 ...  2.06934963933   1.14626069677  0.785186892817  0.764795972809  0
9160  DECOY_592_run0  0  1     -1.692134624  0  0.958494941394  0.00630066071793  -0.328751834466   0.189189195633   -0.482215863506 ...  1.28809100151  0.791833291782  0.803144337956  0.768010655265  0
9161  DECOY_592_run0  0  1   -1.84945466301  0  0.633835911751  0.00561950820789   0.371593955459              0.0   -0.371771330281 ...  2.61135263418   2.02039942856   0.55597268955  0.330880632506  0
9162  DECOY_592_run0  0  1   -2.10957129817  0  0.906453510125  0.00527893195286    0.22376248428  0.0645161271095   -0.255514614107 ...  1.07777568371  0.598394985285  0.893420898864  0.890335674837  0
9163  DECOY_592_run0  0  1   -2.31560973872  0  0.905198852221   0.0028948981677   0.137664821691              0.0    0.876978922737 ...  3.43237660856   2.17985387492  0.539929223558   0.44766613662  0
9164  DECOY_592_run0  0  1   -2.53264045217  0  0.980351785819  0.00664123697296  0.0745622896143   0.205128207803    0.216855907728 ...  1.87132796282  0.908604380766  0.736385791197  0.708796452491  0
9165  DECOY_592_run0  0  1   -3.44062881815  0            -1.0  0.00102172876507  0.0339463918184   0.333333343267   0.0837371951856 ...            1.0             1.0             0.0             0.0  0

                  20              21               22   23  24
9156  -3.49780750275  0.979526073734  0.0632768781482  1.0  11
9157  -1.00938773155  0.860200595908  0.0560026429628  1.0   2
9158  -4.97128486633  0.979526073734  0.0632768781482  1.0  14
9159  -4.11790084839  0.979526073734  0.0632768781482  1.0  12
9160  -2.77628207207  0.979526073734  0.0632768781482  1.0   9
9161  -1.98261642456  0.979526073734  0.0632768781482  1.0   5
9162  -3.34569406509  0.979526073734  0.0632768781482  1.0  10
9163  -5.20470523834  0.979526073734  0.0632768781482  1.0  16
9164  -5.07580375671  0.979526073734  0.0632768781482  1.0  15
9165  -4.78132820129  0.979526073734  0.0632768781482  1.0  13

[10 rows x 25 columns]
pyprophet score --in=test_data.txt --apply_weights=test_data_ms2_weights.csv --random_seed=42
 config settings:
     final_statistics.emp_p: False
     final_statistics.lambda: 0.4
     final_statistics.lfdr_adj: 1.5
     final_statistics.lfdr_eps: 1e-08
     final_statistics.lfdr_monotone: True
     final_statistics.lfdr_transf: probit
     final_statistics.lfdr_trunc: True
     final_statistics.pfdr: False
     final_statistics.pi0_method: smoother
     final_statistics.pi0_smooth_df: 3
     final_statistics.pi0_smooth_log_pi0: False
     group_id: group_id
     is_test: False
     num_processes: 1
     random_seed: 42
     semi_supervised_learner.initial_fdr: 0.15
     semi_supervised_learner.initial_lambda: 0.4
     semi_supervised_learner.iteration_fdr: 0.02
     semi_supervised_learner.iteration_lambda: 0.4
     semi_supervised_learner.num_iter: 10
     tric_chromprob: False
     xeval.fraction: 0.5
     xeval.num_iter: 10
 apply weights
 Data set contains 387 decoy and 387 target groups.
 summary input file:
    9165 lines
    774 transition groups
    17 scores including main score
 start application of pretrained weights
 finished pretrained scoring
 Data set contains 387 decoy and 387 target groups.
 mean q_value = 5.515478e-02, std_dev q_value = 1.675134e-02
 mean s_value = 9.795502e-01, std_dev s_value = 1.132123e-01
 calculated scoring and statistics
 processing input data finished
==================================================================================================
   qvalue        pvalue    svalue           pep           fdr       fnr           fpr          tp  \
0    0.00  2.087219e-13  0.016575  6.248457e-09  8.696747e-13  0.934383  2.087219e-13    6.000000
1    0.01  1.365513e-01  1.000000  6.271745e-01  9.301858e-03  0.000000  1.365513e-01  363.586218
2    0.02  3.019968e-01  1.000000  1.000000e+00  2.035019e-02  0.000000  3.019968e-01  363.450081
3    0.05  7.633976e-01  1.000000  1.000000e+00  4.983013e-02  0.000000  7.633976e-01  363.915059
4    0.10  9.795261e-01  1.000000  1.000000e+00  6.327688e-02  0.000000  9.795261e-01  362.511848
5    0.20           NaN       NaN           NaN           NaN       NaN           NaN         NaN
6    0.30           NaN       NaN           NaN           NaN       NaN           NaN         NaN
7    0.40           NaN       NaN           NaN           NaN       NaN           NaN         NaN
8    0.50           NaN       NaN           NaN           NaN       NaN           NaN         NaN

          tn            fp          fn    cutoff
0  25.000000  5.218048e-12  356.000000  7.249766
1  21.586218  3.413782e+00   -1.586218  1.095946
2  17.450081  7.549919e+00   -1.450081  0.518666
3   5.915059  1.908494e+01   -1.915059 -0.717274
4   0.511848  2.448815e+01   -0.511848 -2.044058
5        NaN           NaN         NaN       NaN
6        NaN           NaN         NaN       NaN
7        NaN           NaN         NaN       NaN
8        NaN           NaN         NaN       NaN
==================================================================================================
WRITTEN: test_data_ms2_summary_stat.csv
WRITTEN: test_data_ms2_full_stat.csv
WRITTEN: test_data_ms2_scored.tsv
WRITTEN: test_data_ms2_report.pdf
TIME: 0 seconds and 128 msecs wall time

